### PR TITLE
diff: remove testify/assert

### PR DESCRIPTION
**Pull request by alrs**
This removes `"github.com/stretchr/testify/assert"` from the `diff` package.

### CODE REVIEW

File: diff/keys_test.go [hunk 1]
```diff
@@ -51,7 +56,9 @@ func TestDiff(t *testing.T) {
 	}
 
 	result := CompareConfigs(g1, g2, opts...)
-	assert.Equal(t, expected, result)
+	if !reflect.DeepEqual(expected, result) {
```
Comment (urso) [inline diff/keys_test.go:59]: here we might prefer go-cmp over reflect.DeepEqual :)

File: diff/keys_test.go [hunk 2]
```diff
@@ -39,10 +40,14 @@ func TestDiff(t *testing.T) {
 	}
 
 	g1, err := ucfg.NewFrom(oneGraph, opts...)
-	assert.NoError(t, err)
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (urso) [inline diff/keys_test.go:45]: this pattern we have quite often. How about:

```
func mustNewFrom(in interface{}, opts ...ucfg.Option) *ucfg.Config {
  c, err := ucfg.NewFrom(in, opts...)
  if err != nil {
    t.Fatalf("Failed to create new config for: %v\nreason:%v", spew.Sdump(in), err)
  }
  return c
}
```

File: diff/keys_test.go [hunk 3]
```diff
@@ -65,14 +72,22 @@ func TestConfigurationWithAddedCompareConfigs(t *testing.T) {
...
 
 	d := CompareConfigs(g1, g2, opts...)
-	assert.True(t, d.HasChanged())
-	assert.True(t, d.HasKeyAdded())
+	if !d.HasChanged() {
+		t.Fatal("expected Diff.HashChanged() to be true")
+	}
```
Comment (urso) [inline diff/keys_test.go:87]: This reminds me that `assert`, and `require` package allow you to add custom messages. For example:
`assert.True(t, d.HasChanged(), "expected Diff.HasChanged() to be true")`.

All the if conditions make me wonder if we want more test helpers like `func expectChanged(t *testing.T, d Diff) { ... }`. Then the test becomes: `expectedChanged(t, CompareConfigs(g1, g2, opts...))`.

**Comment by urso** [comment]
regarding 'utility' functions, one of the recent PRs introduced the `cfgtest` package.

File: diff/keys_test.go (hunk 2 above)
Comment (alrs) [reply diff/keys_test.go]: Done, and pushed.

File: diff/keys_test.go (hunk 1 above)
Comment (alrs) [reply diff/keys_test.go:60]: I'm really loathe to bring in things out of the standard library, `spew` is my only real vice. Are you OK with just using `reflect`?

File: diff/keys_test.go (hunk 3 above)
Comment (alrs) [reply diff/keys_test.go:81]: It's deceptively repetitive.

`!d.HasChanged()` three times
`d.HasChanged()` once.
`!d.HasKeyRemoved()` once.
`d.HasKeyRemoved()` once.

**Comment by urso** [comment]
Hi, his PR is stalled for quite some time. We will close it for now.

There are more changes over time and more to come (hopefully by others) and testify is still ubiquitious. Removing every mention of testify takes quite some effort. Although personally I favor a non-testify approach as you attempted to do here, for sake of consistency in this project and between projects maintained by others I think we will stick with testify.

Thanks for discussions and contributions you have made in this and other PRs and issues!

**Pull request closed without merging**